### PR TITLE
Add cmd for listing all module account and name pairs

### PR DESCRIPTION
**Pull request by mattverse**
## Description
This PR is related to issue https://github.com/osmosis-labs/osmosis/issues/385.

This PR adds a CLI for querying all module accounts, module-accounts, in the auth module.
Using this command would display all the account information, including human readable name of the module, module account address, account number, permissions, etc.

This command would be especially useful for developers using cosmos-sdk as there are lots of instances where a developer would have to look into the module account, but only to do so by getting the module account address after getting the module name, hashing it and bech32fying it in the status quo. By using this command, users would be able to get a quick look on the list of all module accounts' information.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ValarDragon** [comment]
Can this be rebased against `v0.42.9-osmo-rebase` 😅

**Comment by mattverse** [comment]
@ValarDragon Haha Sure! Also, do we have plans any time soon to use 0.43.0 version of cosmos sdk? I also committed this cli command to the cosmos-sdk original repo itself and the commit was included in the 0.43.0 version so just wondering

File: docs/core/proto-docs.md [hunk 1]
```diff
@@ -832,6 +834,31 @@ QueryAccountResponse is the response type for the Query/Account RPC method.
 
 
 
+<a name="cosmos.auth.v1beta1.QueryModuleAccountsRequest"></a>
+
+### QueryModuleAccountsRequest
+QueryModuleAccountRequest is the request type for the Query/Params RPC method.
```
Comment (antstalepresh) [inline docs/core/proto-docs.md:840]: Can this description to be accurate? `QueryModuleAccountsRequest is XXX`

File: docs/core/proto-docs.md [hunk 2]
```diff
@@ -832,6 +834,31 @@ QueryAccountResponse is the response type for the Query/Account RPC method.
...
+
+
+
+
+
+
+<a name="cosmos.auth.v1beta1.QueryModuleAccountsResponse"></a>
+
+### QueryModuleAccountsResponse
+QueryModuleAccountResponse is the response type for the Query/Params RPC method.
```
Comment (antstalepresh) [inline docs/core/proto-docs.md:850]: Can this description to be accurate? `QueryModuleAccountResponse is XXX`

File: x/auth/keeper/grpc_query.go [hunk 3]
```diff
@@ -52,3 +52,28 @@ func (ak AccountKeeper) Params(c context.Context, req *types.QueryParamsRequest)
...
+
+// ModuleAccount returns the account of the module
+func (ak AccountKeeper) ModuleAccounts(c context.Context, req *types.QueryModuleAccountsRequest) (*types.QueryModuleAccountsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	var a []*codectypes.Any
```
Comment (antstalepresh) [inline x/auth/keeper/grpc_query.go:64]: variable name `a` seems too simple name

File: x/auth/client/cli/query.go [hunk 4]
```diff
@@ -203,3 +204,29 @@ func QueryTxCmd() *cobra.Command {
 
 	return cmd
 }
+
```
Comment (antstalepresh) [inline x/auth/client/cli/query.go:207]: Maybe should add CLI test for `QueryModuleAccountsCmd`

**Comment by ValarDragon** [comment]
Yeah we should upgrade to cosmos SDK v0.43.0! If this is already included in there, happy to just wait until we upgrade next version to that, and have this land that way

**Comment by mattverse** [comment]
Sure! That would be the best way to implement this on our side as well! Closing this PR!

**Pull request closed without merging**